### ISSUE TITLE
1.14 Window Hangs Upon Opening

### ISSUE THREAD

**Issue by ndneighbor** [Microsoft/vscode#30438]
![image](https://user-images.githubusercontent.com/15745278/28094826-e54cf8be-6653-11e7-8c29-919e91d90d36.png)

- VSCode Version: 1.14.0
- OS Version: Mac OS Sierra 10.12.5
Steps to Reproduce:

1. Upgrade to 1.14
2. Use the editor for a few minutes

Reproduces without extensions: Yes

At the moment I am gathering more data as to what extensions it might be colliding with, when I run
it with `code --disable-extensions` it starts, but when I open a project it still crashes. The bug only affects the window, I can still interact with the running process by successfully getting the about window to display.

**Comment by ndneighbor** [comment]
![image](https://user-images.githubusercontent.com/15745278/28095285-7edfc392-6656-11e7-8cbf-3ffba3384ab0.png)

`code --disable-extensions` allows me to open a project but still crashes upon opening it

**Comment by devtrix** [comment]
Same here

**Comment by tuxagon** [comment]
On Ubuntu 16.10, I have the same issue; however, `code --disable-extensions` will allow me to interactive with vscode.

**Comment by jilustrisimo** [comment]
same here, OSX 10.12.5

**Comment by chrmarti** [comment]
@ndneighbor Could you clarify if `code --disable-extensions /path/to/project` works?

@ndneighbor @devtrix @tuxagon Could you append the list of installed extensions here? `code --list-extensions`? Also: Does this happen immediately after startup or only after a while and when doing something specific?

**Comment by DServy** [comment]
Failing on Typescript projects, and also welcome screen becomes unresponsive on start up as well.

**Comment by chrmarti** [comment]
@DServy Does this reproduce with `code --disable-extensions`?

**Comment by DServy** [comment]
@chrmarti Seems to work with that flag enabled.

List of extentions:

- CraigMaslowski.erb
- DavidHelmer.mavensmate
- EditorConfig.EditorConfig
- atefth.ruby-on-rails-snippets
- bengreenier.vscode-node-readme
- casualjim.theme-desertex
- christian-kohler.npm-intellisense
- chuckjonas.apex-autocomplete
- chuckjonas.apex-debug
- chuckjonas.apex-pmd
- codezombiech.gitignore
- cometeer.spacemacs
- dbaeumer.vscode-eslint
- donjayamanne.githistory
- dweizhe.docthis-customize-tags
- eg2.tslint
- eg2.vscode-npm-script
- fknop.vscode-npm
- flowtype.flow-for-vscode
- hoovercj.ruby-linter
- jasonnutter.github-pr
- ms-mssql.mssql
- ms-vscode.sublime-keybindings
- msjsdiag.debugger-for-chrome
- rebornix.Ruby
- robertohuertasm.vscode-icons
- searKing.preview-vscode
- shyykoserhiy.vscode-spotify
- vncz.vscode-apielements
- vsmobile.vscode-react-native
- xabikos.JavaScriptSnippets
- xabikos.ReactSnippets
- zxh404.vscode-proto3

**Comment by devtrix** [comment]
@chrmarti 
This extension was my culprit atefth.ruby-on-rails-snippets-0.0.1

**Comment by rebornix** [comment]
if you remove rails 4 snippet extension, the issue can be gone.

**Comment by devtrix** [comment]
@DServy 
Try removing this extension atefth.ruby-on-rails-snippets-0.0.1

**Comment by DServy** [comment]
@devtrix @chrmarti @rebornix Great catch, thanks again!

**Comment by jilustrisimo** [comment]
@devtrix It worked, thanks.

**Comment by chrmarti** [comment]
Duplicate of #30407. Thanks.
